perlmod/Fink/PkgVersion.pm

```diff
@@ -5415,52 +5752,64 @@ sub get_perl_dir_arch {
-			} elsif ($perlversion eq  "5.18.2" and Fink::Services::get_kernel_vers() eq '15') {
+			} elsif ((&version_cmp($perlversion, '=', "5.18.2")) and Fink::Services::get_kernel_vers() == '15') {
 				# 10.11 system-perl is 5.18.2, but the only supplied
 				# interpreter is /usr/bin/perl5.18 (not perl5.18.2)
 				$perlcmd = "/usr/bin/arch -%m perl5.18";
-			} elsif ($perlversion eq  "5.18.2" and Fink::Services::get_kernel_vers() eq '16') {
+			} elsif ((&version_cmp($perlversion, '=', "5.18.2")) and Fink::Services::get_kernel_vers() == '16') {
 				# 10.12 system-perl is 5.18.2, but the only supplied
 				# interpreter is /usr/bin/perl5.18 (not perl5.18.2)
 				$perlcmd = "/usr/bin/arch -%m perl5.18";
-			} elsif ($perlversion eq  "5.18.2" and Fink::Services::get_kernel_vers() eq '17') {
+			} elsif ((&version_cmp($perlversion, '=', "5.18.2")) and Fink::Services::get_kernel_vers() == '17') {
 				# 10.13 system-perl is 5.18.2, but the only supplied
 				# interpreter is /usr/bin/perl5.18 (not perl5.18.2)
 				$perlcmd = "/usr/bin/arch -%m perl5.18";
-			} elsif ($perlversion eq  "5.18.2" and Fink::Services::get_kernel_vers() eq '18') {
-				# 10.14 system-perl is 5.18.2, but the only supplied
+			} elsif ((&version_cmp($perlversion, '=', "5.18.2")) and Fink::Services::get_kernel_vers() == '18' and Fink::Services::get_kernel_vers_minor() <= '5') {
+				# 10.14.[0-4] system-perl is 5.18.2, but the only supplied
 				# interpreter is /usr/bin/perl5.18 (not perl5.18.2)
 				$perlcmd = "/usr/bin/arch -%m perl5.18";
-			} elsif ($perlversion eq  "5.18.4" and Fink::Services::get_kernel_vers() eq '18') {
+			} elsif ((&version_cmp($perlversion, '=', "5.18.4")) and Fink::Services::get_kernel_vers() == '18') {
 				# 10.14.5 system-perl is 5.18.4, but the only supplied
 				# interpreter is /usr/bin/perl5.18 (not perl5.18.4)
 				$perlcmd = "/usr/bin/arch -%m perl5.18";
-			} elsif ($perlversion eq  "5.18.4" and Fink::Services::get_kernel_vers() eq '19') {
+			} elsif ((&version_cmp($perlversion, '=', "5.18.4")) and Fink::Services::get_kernel_vers() == '19') {
 				# 10.15 system-perl is 5.18.4, but the only supplied
 				# interpreter is /usr/bin/perl5.18 (not perl5.18.4)
 				$perlcmd = "/usr/bin/arch -%m perl5.18";
+			} elsif ((&version_cmp($perlversion, '=', "5.28.2")) and Fink::Services::get_kernel_vers() == '20' and Fink::Services::get_kernel_vers_minor() <= '3') {
+				# 11.0.[0-2] system-perl is 5.28.2, but the only supplied
+				# interpreter is /usr/bin/perl5.28 (not perl5.28.2)
+				$perlcmd = "/usr/bin/arch -%m perl5.28";
+			} elsif ((&version_cmp($perlversion, '=', "5.30.2")) and Fink::Services::get_kernel_vers() == '20') {
+				# 11.3 system-perl is 5.30.2, but the only supplied
+				# interpreter is /usr/bin/perl5.30 (not perl5.30.2)
+				$perlcmd = "/usr/bin/arch -%m perl5.30";
+			} elsif ((&version_cmp($perlversion, '=', "5.30.3")) and Fink::Services::get_kernel_vers() == '21') {
+				# 12.0 system-perl is 5.30.3, but the only supplied
```

Comment:
```suggestion
			} elsif ((&version_cmp($perlversion, '=', "5.30.3")) and Fink::Services::get_kernel_vers() ge '21') {
				# 12.0/13.0 system-perl is 5.30.3, but the only supplied
```
The present version creates a perl `%{default_script}`
```
ARCHFLAGS="-l/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/System/Library/Perl/5.30/darwin-thread-multi-2level/CORE" /usr/bin/arch -arm64 perl5.30.3 Makefile.PL PERL="/usr/bin/arch -arm64 perl5.30.3" PREFIX=/opt/sw2 INSTALLPRIVLIB=/opt/sw2/lib/perl5/5.30.3 INSTALLARCHLIB=/opt/sw2/lib/perl5/5.30.3/darwin-thread-multi-2level INSTALLSITELIB=/opt/sw2/lib/perl5/5.30.3 INSTALLSITEARCH=/opt/sw2/lib/perl5/5.30.3/darwin-thread-multi-2level INSTALLMAN1DIR=/opt/sw2/share/man/man1 INSTALLMAN3DIR=/opt/sw2/share/man/man3 INSTALLSITEMAN1DIR=/opt/sw2/share/man/man1 INSTALLSITEMAN3DIR=/opt/sw2/share/man/man3 INSTALLBIN=/opt/sw2/bin INSTALLSITEBIN=/opt/sw2/bin INSTALLSCRIPT=/opt/sw2/bin INSTALLSITESCRIPT=/opt/sw2/bin
```
failing on missing `perl5.30.3`.

Reply:
No.
```
% uname -p 
i386
% uname -m
x86_64
```
This did work with this branch up to 2 January.

Reply:
But the perl failure is unrelated to the architecture, at least my 22.2.0 system only has
/usr/bin/perl5.30, /usr/bin/perl5.34

Reply:
still not going to fix it, if native works there is no point

Reply:
thanks for reporting it, it's fixed now

Reply:
It does not, there is no working `/usr/bin/arch -arm64 perl5.30.3`. (Could not even go as far as testing this with Rosetta due to the bootstrap failure).
If this does exist on some other systems, I am of course open to try and fix my Ventura installation.

Reply:
I mean I could add an other block and move 13.x to perl5.34 but that would make an other pile of issues since no modules would support 5.34.0 in the dist.

Reply:
also for the rosetta you didn't post a single actionable error, so even if I wanted to help I couldn't.  Since the M1 I'm using it's against policy to install Rosetta (aka it's locked and I can't) so I can't test it to find the errors myself.  My guess is that in some cases on configures it's checking processor info and reporting arm/arm64.   But since I can't see anything it's just a guess.  That being said, the perl issue should be sorted which is what I was focused on.  If you want to fight the rosetta battle that's fine but I'll need to see actual errors to help.

Reply:
> It does not, there is no working `/usr/bin/arch -arm64 perl5.30.3`. (Could not even go as far as testing this with Rosetta due to the bootstrap failure).

Sorry, that referred to your "if native works there is no point" – the patch just came in simultaneously with my comment. Thanks, building perlmods now works!
And most things work with 5.30 (some packages have ancient perl that is already illegal in 5.30, so certainly no point in moving further to 5.34 yet).

Reply:
The first suspicious error is after building libiconv when install/configuration fails on `libiconv depends on darwin (>= 22-1)`. Though both the M1 installation and the bootstrapped Rosetta have `darwin 22.2.0-1`.
But apparently that killed that stage, as all packages that should have been built afterwards, from lzma, ncurses to xz, are missing. I certainly don't see anything in the latest libiconv updates that I could relate to this; perhaps the updates to cputable/tupletable?